src/Aspire.Hosting/ApplicationModel/ResourceNotificationService.cs

```diff
@@ -716,6 +718,27 @@ static ResourceCommandSnapshot CreateCommandFromAnnotation(ResourceCommandAnnota
         }
     }
 
+    /// <summary>
+    /// Use icon annotations to update resource snapshot.
+    /// </summary>
+    private static CustomResourceSnapshot UpdateIcons(IResource resource, CustomResourceSnapshot previousState)
+    {
+        var iconAnnotation = resource.Annotations.OfType<ResourceIconAnnotation>().FirstOrDefault();
+        
+        if (iconAnnotation == null)
+        {
+            // No icon annotation, keep existing icon information
+            return previousState;
+        }
+
+        // Update the snapshot with icon information from annotation
+        return previousState with 
+        { 
+            IconName = iconAnnotation.IconName,
+            IconVariant = iconAnnotation.IconVariant
+        };
```

Comment:
Don't overwrite if there is already a value.

Add unit test for this scenario.